client/src/LayoutComponents/AllStudents/AllStudents.js

```diff
@@ -11,15 +11,26 @@ class AllStudents extends Component {
     students: [],
   };
 
-  componentDidMount() {
+    componentDidMount() {
+    this.getStudents();
+   
+  }
+
+  componentDidUpdate(prevProps ,prevState) {
+    const { students } = this.state;
+    if (JSON.stringify(students) !== JSON.stringify(prevState.students) ){
+      this.getStudents();
```

Comment:
I think this will cause the students to be fetched twice, because the first getStudents call will setState(), which in turn will trigger this if-statement.

Reply:
Nevermind, I saw that this code was removed in #142 :)

Reply:
Sorry about it, I was saw this now, thank you